skore/src/skore/sklearn/train_test_split.py

```diff
@@ -0,0 +1,252 @@
+"""Implement train_test_split."""
+
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING, Any, Optional, Union
+
+import pandas
+from numpy.random import RandomState
+
+from skore.project import Project
+
+if TYPE_CHECKING:
+    from skore.sklearn.cross_validate import MLTask
+
+    ArrayLike = Any
+
+
+def _find_ml_task(y) -> MLTask:
+    """Guess the ML task being addressed based on a target array.
+
+    Parameters
+    ----------
+    y : numpy.ndarray
+        A 1-dimensional target vector.
+
+    Returns
+    -------
+    Literal["binary-classification", "multiclass-classification",
+    "regression", "clustering", "unknown"]
+        The guess of the kind of ML task being performed.
+    """
+    import sklearn.utils.multiclass
+
+    if y is None:
+        # NOTE: The task might not be clustering
+        return "clustering"
+
+    type_of_target = sklearn.utils.multiclass.type_of_target(y)
+
+    if type_of_target == "binary":
+        return "binary-classification"
+
+    if type_of_target == "multiclass":
+        return "multiclass-classification"
+
+    if "continuous" in type_of_target:
+        return "regression"
+
+    if type_of_target == "unknown":
+        return "unknown"
+
+    return "unknown"
+
+
+class TrainTestSplitWarning(Warning):
+    """Interface for a train-test-split warning."""
+
+    MSG: str
+
+    @staticmethod
+    def check(*args, **kwargs) -> bool:
+        """Perform the check.
+
+        Returns
+        -------
+        bool
+            True if the check passed, False otherwise.
+        """
+        ...
```

Comment:
To prepare the future iterations and to increase readability, how about splitting the codes something like:

```
sklearn/
|- cross_validate.py
|- train_test_split/
   | - __init__.py
   | - train_test_split.py
   | - warning/
      | - __init__.py
      | - interface.py
      | - first_warning_class.py
      | - second_warning_class.py
```

Reply:
done